src/vs/workbench/contrib/chat/browser/chatMarkdownRenderer.ts

```diff
@@ -100,6 +100,7 @@ export class ChatMarkdownRenderer extends MarkdownRenderer {
 		// In some cases, the renderer can return text that is not inside a <p>,
 		// but our CSS expects text to be in a <p> for margin to be applied properly.
 		// So just normalize it.
+		result.element.normalize();
```

Comment:
I'll create a separate PR for September that looks into this code. I think we actually need to do this replacement for all children, not just the last one. Otherwise you can still end up with top level text nodes